Explorer/Assets/DCL/SDKComponents/AudioSources/Systems/UpdateAudioSourceSystem.cs

```diff
@@ -122,5 +123,20 @@ private void HandleSDKChanges(PBAudioSource sdkComponent, ref AudioSourceCompone
 
             sdkComponent.IsDirty = false;
         }
+
+        /// <summary>
+        /// Mutes an AudioSource depending on whether its world position in contained in the current scene.
+        /// If the AudioSource is in the bounding box of the scene, it is un-muted; otherwise it is muted.
+        /// </summary>
+        /// <param name="entity">The entity that contains the AudioSource and its Transform.</param>
+        /// <param name="component">The AudioSource component.</param>
+        [Query]
+        private void UpdateAudioSourceMute(in Entity entity, ref AudioSourceComponent component)
+        {
+            Vector3 audioWorldPosition = World.Get<TransformComponent>(entity).Transform.position;
+            component.Mute(audioWorldPosition.y > sceneData.Geometry.Height
+                            || audioWorldPosition.y < 0.0f
+                            || !sceneData.Geometry.CircumscribedPlanes.Intersects(audioWorldPosition));
```

Comment:
Maybe it's better to just use the `sceneStateProvider.IsCurrent` to check if the main player is inside the audio scene, instead of doing all the scene bounds calculations, as we do [here](https://github.com/decentraland/unity-explorer/blob/0594ed297b17030ff167968c277baf9a77e3a362/Explorer/Assets/DCL/Interaction/Raycast/Systems/ExecuteRaycastSystem.cs#L87).

Reply:
But it does not check the main player only, it checks all entities (with an AudioSource). In fact, players (avatars, emotes) are not affected by this feature.

Reply:
Because using the `CircumscribedPlanes` won't work completely accurate for cases like "irregular shaped scenes" ([from what I see the CircumscribedPlanes is created only using the rectangular "outer bounds" of the scene](https://github.com/decentraland/unity-explorer/blob/main/Explorer/Assets/Scripts/Utility/ParcelMathHelper.cs#L57~L65)) 

e.g. in this kind of scene the audio could still be playing outside the real bounds of the scene, but inside its outer bounds:
https://adr.decentraland.org/resources/ADR-192/scene-inner-bounds.png
(taken from https://adr.decentraland.org/adr/ADR-192)

That's why I suggested originally in the issue that as a first iteration it would be better to just mute everything outside the current scene (you can check with `sceneStateProvider.IsCurrent`) otherwise the scene bounds checks start getting more complex for the case of irregular shaped scenes

Reply:
Ok, I read the doc but I thought it was not implemented yet after seeing how `CheckColliderBoundsSystem` was implmented. Will take a look to that then and change the implementation, thank you.

Reply:
> But it does not check the main player only

Sorry what I meant is that by using `sceneStateProvider.IsCurrent` you can check if the main player is inside the scene in which the system is running or not.

Reply:
> I read the doc but I thought it was not implemented yet

You are right, those complex checks are not implemented, and if we want to avoid having that complexity for now we can avoid checking that the audio entities are inside/outside each scene, and just check if the main player is inside the scene in which the audio system is running, and if it's not then we can mute all the audio entities since they are playing on a scene that is no the one the main player is in.

It's a suggestion to make an easier 1st iteration to tackle the problem, we can do the more complex solution in the future

Reply:
What if there are entities that belong to the current scene but are placed outside (or move in and out)? Maybe I should do both things, depending on IsCurrent.

Reply:
if we care about that then yes we need the full complex check with several layers (first only checking outer bounds and if the object is inside then check parcel-wise).

But I think at this first iteration on the problem it's OK to have audio of the current scene playing outside it, it will bother no one since the player is standing inside the scene that owns those audio entities right? and if they get ouf of the scene, those audio objects will be muted.

The main problematic is not bothering other scenes by having audio that doesn't belong to the current scene, playing inside the current scene.

Maybe @m3taphysics has a different input on this and can join us to help define the final behaviour we want =)

Reply:
I have added both approaches.

Reply:
I definitely recommend using `IsCurrent` - it's precise as it uses parcels checks. You can check `UpdateCurrentSceneSystem` to see how it works.
`Circumscribed planes` in this case are fully redundant, moreover it can produce additional overhead to check

Reply:
I think we can accept that, since it will only happen when the user is actually inside the scene, so it's not bothering users on other scenes

Reply:
I have finally removed the part that checks for the boundaries. Now it only uses the IsCurrent flag.